src/test/java/dev/openfeature/sdk/providers/memory/InMemoryProviderTest.java

```diff
@@ -0,0 +1,159 @@
+package dev.openfeature.sdk.providers.memory;
+
+import com.google.common.collect.ImmutableMap;
+import dev.openfeature.sdk.*;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static dev.openfeature.sdk.providers.memory.InMemoryProvider.mapToStructure;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class InMemoryProviderTest {
+
+    private static Client client;
+
+    private static final AtomicBoolean isProviderReady = new AtomicBoolean(false);
+
+    private static final AtomicBoolean isConfigurationChanged = new AtomicBoolean(false);
+
+    @SneakyThrows
+    @BeforeAll
+    static void beforeAll() {
+        Map<String, Flag<?>> flags = buildFlags();
+        InMemoryProvider provider = new InMemoryProvider(flags);
+        OpenFeatureAPI.getInstance().setProvider(provider);
+
+        // TODO: setProvider with wait for init, pending https://github.com/open-feature/ofep/pull/80
+        Thread.sleep(500);
+
+        client = OpenFeatureAPI.getInstance().getClient();
+
+        client.onProviderReady(eventDetails -> isProviderReady.set(true));
+
+        client.onProviderConfigurationChanged(eventDetails -> isConfigurationChanged.set(true));
+        provider.updateFlags(flags);
+    }
+
+    public static Map<String, Flag<?>> buildFlags() {
```

Comment:
Given the public accessor status & multiple usages, I think this fits well into testutils - https://github.com/open-feature/java-sdk/tree/main/src/test/java/dev/openfeature/sdk/testutils

Probably with a class named `TestFlags`